### PR TITLE
Improve autosave dialog and behavior

### PR DESCRIPTION
**Pull request by ClementPasteau**
Also improve all confirmation dialogs with clearer button labels

Improve autosave popup to prevent 2 things:
- displaying the warning about a newer autosave version if the user has only clicked on preview without making any changes
- displaying the warning when a project is unzipped with an autosave

Also improve the dialogs design & wording.
image

### CODE REVIEW

File: newIDE/app/src/MainFrame/index.js [hunk 1]
```diff
@@ -841,107 +841,112 @@ const MainFrame = (props: Props) => {
...
       };
 
       setIsLoadingProject(true);
 
       // Try to find an autosave (and ask user if found)
-      return delay(150)
-        .then(() => checkForAutosave())
-        .then(fileMetadata => onOpen(fileMetadata, setLoaderModalProgress))
-        .catch(err => {
+      try {
```
Comment (4ian) [inline newIDE/app/src/MainFrame/index.js:886]: 🙏

File: newIDE/app/src/MainFrame/index.js [hunk 2]
```diff
@@ -841,107 +841,112 @@ const MainFrame = (props: Props) => {
...
-              i18n._(errorMessage),
-            ].join('\n'),
-            errorId: 'project-open-error',
-            rawError: error,
-          });
-          setIsLoadingProject(false);
-          setLoaderModalProgress(null, null);
-          return Promise.reject(error);
+          serializedProject.delete();
+          return Promise.resolve(state);
```
Comment (4ian) [inline newIDE/app/src/MainFrame/index.js:922]: nit: might not need the Promise.resolve?

**Review by 4ian — APPROVED** [review]

File: newIDE/app/src/UI/Alert/ConfirmDeleteDialog.js [hunk 3]
```diff
@@ -15,9 +15,11 @@ type Props = {|
...
   fieldMessage: MessageDescriptor,
-  confirmText: string,
+  confirmHintText: string,
   onConfirm: () => void,
   onDismiss: () => void,
+  confirmText?: MessageDescriptor,
+  dismissText?: MessageDescriptor,
 |};
 
 function ConfirmDeleteDialog(props: Props) {
```
Comment (AlexandreSi) [inline newIDE/app/src/UI/Alert/ConfirmDeleteDialog.js:25]: canConfirm must use confirmHintText. I'm not sure confirmHintText is a great wording. I think it should stay confirmText and the labels should be `confirmButtonLabel` and `DismissButtonLabel`

File: newIDE/app/src/MainFrame/index.js (hunk 2 above)
Comment (ClementPasteau) [reply newIDE/app/src/MainFrame/index.js:922]: oopsie

File: newIDE/app/src/UI/Alert/ConfirmDeleteDialog.js (hunk 3 above)
Comment (ClementPasteau) [reply newIDE/app/src/UI/Alert/ConfirmDeleteDialog.js:25]: Yeah you're probably right, I'll update that

Comment (ClementPasteau) [reply newIDE/app/src/UI/Alert/ConfirmDeleteDialog.js:25]: done!

**Pull request merged**